cli/src/torii/routing.rs

```diff
@@ -423,16 +425,38 @@ async fn handle_metrics(sumeragi: Arc<Sumeragi>, _network: Addr<IrohaNetwork>) -
 }
 
 #[cfg(feature = "telemetry")]
-#[allow(clippy::unused_async)] // TODO: remove?
-async fn handle_status(sumeragi: Arc<Sumeragi>, _network: Addr<IrohaNetwork>) -> Result<Json> {
-    // TODO: remove network
+#[allow(clippy::unnecessary_wraps)]
+fn handle_status(sumeragi: &Sumeragi) -> Result<warp::reply::Json, Infallible> {
     if let Err(error) = sumeragi.update_metrics() {
         iroha_logger::error!(%error, "Error while calling `sumeragi::update_metrics`.");
     }
     let status = Status::from(&sumeragi.metrics());
     Ok(reply::json(&status))
 }
 
+#[cfg(feature = "telemetry")]
+#[allow(clippy::unused_async)]
+async fn handle_status_precise(sumeragi: Arc<Sumeragi>, segment: String) -> Result<Json> {
+    if let Err(error) = sumeragi.update_metrics() {
+        iroha_logger::error!(%error, "Error while calling `sumeragi::update_metrics`.");
+    }
+    // TODO: This probably can be optimised to elide the full
+    // structure. Ideally there should remain a list of fields and
+    // field aliases somewhere in `serde` macro output, which can
+    // elide the creation of the value, and directly read the value
+    // behind the mutex.
+    let status = Status::from(&sumeragi.metrics());
+    match serde_json::to_value(status) {
+        Ok(value) => Ok(value
+            .get(segment)
+            .map_or_else(|| reply::json(&value), reply::json)),
+        Err(err) => {
+            iroha_logger::error!(%err, "Error while converting to JSON value");
+            Ok(reply::json(&None::<String>))
```

Comment:
interesting, I didn't know you can add generic parameter to enum variant?

Reply:
Normally you don't need to.